pkg/cmd/admin/groups/sync/syncerror/handler.go

```diff
@@ -0,0 +1,114 @@
+package syncerror
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+)
+
+// Handler knows how to handle errors
+type Handler interface {
+	// HandleError processess an error without mutating it. If the error is determined to be fatal,
+	// a non-nil error should be returned.
+	HandleError(err error) (handled bool, fatalError error)
+}
+
+func NewCompoundHandler(handlers ...Handler) Handler {
+	return &compoundHandler{handlers: handlers}
+}
+
+// compoundHandler chains other error handlers
+type compoundHandler struct {
+	handlers []Handler
+}
+
+// HandleError asks each of the handlers to handle the error. If any handler decides the error is fatal or handles it,
+// the chain is broken.
+func (h *compoundHandler) HandleError(err error) (bool, error) {
+	var handled bool
+	var handleErr error
+	for _, handler := range h.handlers {
+		handled, handleErr = handler.HandleError(err)
+		if handled || handleErr != nil {
+			break
+		}
+	}
+
+	return handled, handleErr
```

Comment:
return `false, nil` here to be clearer about what happens if no handlers handle it